### DISCUSSION TITLE
rye lock is pretty slow, anything I can do to improve it or debug?

### DISCUSSION THREAD

**Discussion by indigoviolet** [astral-sh/rye#495] [Q&A]
I have a fairly middle-of-the-road workspace with a couple of projects, but every time I need to build lockfiles it is so slow. It seems like it runs through "Collecting ..." lists 4x, 2x for prod and 2x for dev dependencies. Within prod, one time for dependencies and one time for local projects + dependencies. 

Am I doing something wrong?

If it's useful, my workspace pyproject.toml has no dependencies listed; the projects each have dependencies. 

Happy to provide more logs or config snippets, just not sure what is useful.

**Comment by ischaojie** [comment]
As far as I can guess, this could be a problem with `pip-tools`, MAYBE. The part about lock-related is something rye desperately needs to fix.

**Comment by mitsuhiko** [comment]
It's definitely an issue and it's something I'm trying to resolve in one way or another.

**Comment by bluss** [comment]
For workspace-only projects, I put all dependencies in dev-dependencies, which is not a nice thing, but it speeds up sync since it avoids some double work. When it's possible, it helps (and ideally, in the future this will not be needed.)
